### PR TITLE
Add permission details

### PR DESCRIPTION
**Pull request by ltseeley**
Adds a display name and description to all checked REST API permissions along with the `/authorization/permissions` endpoint and `splinter permissions` CLI command for viewing all permission details.

# Testing

1. Start splinterd:

```bash
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features=experimental -- database migrate
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features=experimental -- cert generate
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path splinterd/Cargo.toml --features experimental -- --rest-api-endpoint http://localhost:8080 --tls-insecure -vv
```

2. Generate a key pair for yourself you don't already have one:

```bash
cargo run --manifest-path cli/Cargo.toml --features=experimental -- keygen
```

3. Copy your public key to the allow keys file:

```bash
cat ~/.cylinder/keys/<your-username>.pub > /tmp/splinter/data/allow_keys
```

4. List permissions using the CLI:

```bash
cargo run --manifest-path cli/Cargo.toml --features=experimental -- permissions -U http://localhost:8080
ID                              NAME                   DESCRIPTION
registry.read                   Registry read          Allows the client to read the registry
registry.write                  Registry write         Allows the client to modify the registry
circuit.read                    Circuit read           Allows the client to read circuit state
circuit.write                   Circuit write          Allows the client to modify circuit state
scabbard.write                  Scabbard write         Allows the client to submit batches to scabbard services
scabbard.read                   Scabbard read          Allows the client to read scabbard services' state and batch statuses
authorization.maintenance.read  Maintenance mode read  Allows the client to check maintenance mode status
authorization.maintenance.write Maintenance mode write Allows the client to enable/disable maintenance mode
status.read                     Status read            Allows the client to get node status info
authorization.rbac.read         RBAC read              Allows the client to read roles, identities, and role assignments
authorization.rbac.write        RBAC write             Allows the client to modify roles, identities, and role assignments
health.read                     Health read            Allows the client to check node health
authorization.permissions.read  Permissions read       Allows the client to read REST API permissions
```

### CODE REVIEW

File: libsplinter/src/rest_api/auth/authorization/routes/actix.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
+// Copyright 2018-2020 Cargill Incorporated
```
Comment (agunde406) [inline libsplinter/src/rest_api/auth/authorization/routes/actix.rs:1]: Should be updated to be 2021

Comment (agunde406) [reply libsplinter/src/rest_api/auth/authorization/routes/actix.rs:1]: same for other new files

File: splinterd/api/static/openapi.yaml [hunk 2]
```diff
@@ -414,9 +414,12 @@ paths:
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  type: '#/components/schemas/Permission'
+                type: object
```
Comment (agunde406) [inline splinterd/api/static/openapi.yaml:417]: This change does not belong to this commit

File: cli/man/splinter-permissions.1.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  --url http://example.com:8080
+ID,NAME,DESCRIPTION
+registry.read,Registry read,Allows the client to read the registry
+registry.write,Registry write,Allows the client to modify the registry
+circuit.read,Circuit read,Allows the client to read circuit state
+circuit.write,Circuit write,Allows the client to modify circuit state
+...
+```
+
+The following command displays REST API permissions as CSV (output abbreviated):
```
Comment (agunde406) [inline cli/man/splinter-permissions.1.md:83]: CSV -> json

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libsplinter/src/rest_api/auth/authorization/routes/actix.rs (hunk 1 above)
Comment (ltseeley) [reply libsplinter/src/rest_api/auth/authorization/routes/actix.rs]: Fixed

File: splinterd/api/static/openapi.yaml (hunk 2 above)
Comment (ltseeley) [reply splinterd/api/static/openapi.yaml:417]: Fixed (moved to previous commit)

File: cli/man/splinter-permissions.1.md (hunk 3 above)
Comment (ltseeley) [reply cli/man/splinter-permissions.1.md]: Fixed

**Review by vaporos — APPROVED** [review]
Here or in a subsequent PR, the output should be sorted by the ID.

**Review by agunde406 — APPROVED** [review]
